pkg/executables/cmk.go

```diff
@@ -208,15 +219,26 @@ func (c *Cmk) ValidateDomainPresent(ctx context.Context, domain string) (v1alpha
 		return domainIdentifier, fmt.Errorf("failed to parse response into json: %v", err)
 	}
 	domains := response.CmkDomains
-	if len(domains) > 1 {
-		return domainIdentifier, fmt.Errorf("duplicate domain %s found", domain)
-	} else if len(domains) == 0 {
+	if len(domains) == 1 {
+		domainIdentifier.Id = domains[0].Id
+		domainIdentifier.Name = domains[0].Name
+	} else if len(domains) > 1 {
+		domainPath := rootDomain + domainDelimiter + domain
```

Comment:
Use `strings.Join([]string{rootDomain, domain}, domainDelimeter)` to minimize allocations.